internal/controllers/machine/machine_controller_phases_test.go

```diff
@@ -1003,82 +1080,170 @@ func TestReconcileInfrastructure(t *testing.T) {
 					},
 				},
 			},
-			expectResult:  ctrl.Result{},
-			expectError:   false,
-			expectChanged: true,
+			InfraMachineGetError: nil,
+			expectResult:         ctrl.Result{},
+			expectError:          false,
 			expected: func(g *WithT, m *clusterv1.Machine) {
 				g.Expect(m.Status.InfrastructureReady).To(BeTrue())
-				g.Expect(m.GetOwnerReferences()).NotTo(ContainRefOfGroupKind("cluster.x-k8s.io", "MachineSet"))
+				g.Expect(ptr.Deref(m.Spec.ProviderID, "")).To(Equal("test://id-1"))
+				g.Expect(m.Spec.FailureDomain).To(BeNil())
+				g.Expect(m.Status.Addresses).ToNot(BeNil())
 			},
 		},
 		{
-			name: "ready bootstrap, infra, and nodeRef, machine is running, infra object is deleted, expect failed",
+			name:    "infra machine ready and with all the optional fields, it should reconcile and data should surface on the machine",
+			machine: defaultMachine.DeepCopy(),
+			infraMachine: map[string]interface{}{
+				"kind":       "GenericInfrastructureMachine",
+				"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
+				"metadata": map[string]interface{}{
+					"name":      "infra-config1",
+					"namespace": metav1.NamespaceDefault,
+				},
+				"spec": map[string]interface{}{
+					"providerID":    "test://id-1",
+					"failureDomain": "foo",
+				},
+				"status": map[string]interface{}{
+					"ready": true,
+					"addresses": []interface{}{
+						map[string]interface{}{
+							"type":    "InternalIP",
+							"address": "10.0.0.1",
+						},
+						map[string]interface{}{
+							"type":    "InternalIP",
+							"address": "10.0.0.2",
+						},
+					},
+				},
+			},
+			InfraMachineGetError: nil,
+			expectResult:         ctrl.Result{},
+			expectError:          false,
+			expected: func(g *WithT, m *clusterv1.Machine) {
+				g.Expect(m.Status.InfrastructureReady).To(BeTrue())
+				g.Expect(ptr.Deref(m.Spec.ProviderID, "")).To(Equal("test://id-1"))
+				g.Expect(ptr.Deref(m.Spec.FailureDomain, "")).To(Equal("foo"))
+				g.Expect(m.Status.Addresses).ToNot(BeNil())
+			},
+		},
+		{
+			name:    "infra machine ready and no provider ID, it should fail",
+			machine: defaultMachine.DeepCopy(),
+			infraMachine: map[string]interface{}{
+				"kind":       "GenericInfrastructureMachine",
+				"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
+				"metadata": map[string]interface{}{
+					"name":      "infra-config1",
+					"namespace": metav1.NamespaceDefault,
+				},
+				"spec": map[string]interface{}{},
+				"status": map[string]interface{}{
+					"ready": true,
+				},
+			},
+			InfraMachineGetError: nil,
+			expectResult:         ctrl.Result{},
+			expectError:          true,
+		},
+		{
+			name: "should back to infrastructure not ready",
```

Comment:
```suggestion
			name: "should revert back to infrastructure not ready",
```
I don't think it is correct that a machine can revert back to infrastructure not ready, opinions?

Reply:
From ready? It should be possible in some cases, like ec2 status isn't reporting healthy, it could be temporary but should be surfaced

Reply:
We already have the infrastructure ready condition for tracking tracks the status for the infra machine during its entire lifecycle, but IMO the infrastructure ready flag we are using during initial provisioning should never flip back (and this is also one of the key point clarified by the recent proposal about improving status).

Reply:
Fixed, the  `infrastructureReady` flag cannot flip back anymore, and with v1beta2 we are going to rename the field into `Initialization.InfrastructureProvisioned` to avoid any confusion (the `infrastructureReady` condition instead can flip back and forth).

Reply:
I agree this field should never go back to false. The condition should be used instead to signal continuous state